source/framework/vx_array.c

```diff
@@ -22,6 +22,182 @@ static void ownInitArrayObject(
     vx_array arr, vx_enum item_type, vx_size capacity, vx_bool is_virtual);
 static vx_size ownGetArrayItemSize(vx_context context, vx_enum item_type);
 static vx_bool ownIsValidArrayItemType(vx_context context, vx_enum item_type);
+static vx_bool ownIsValidInputAndOutputArrays(vx_array input, vx_array output);
+static vx_status isArrayCopyable(vx_array input, vx_array output);
+static vx_status isArraySwapable(vx_array input, vx_array output);
+static vx_status copyArray(vx_array input, vx_array output);
+static vx_status swapArray(vx_array input, vx_array output);
+static vx_status VX_CALLBACK arrayKernelCallback(vx_enum kernel_enum, vx_bool validate_only, vx_enum optimization, const vx_reference params[], vx_uint32 num_params);
+
+/*! \brief checks that the input and output references are not the same, and that both are valid arrays
+*/
+static vx_bool ownIsValidInputAndOutputArrays(vx_array input, vx_array output)
```

Comment:
GENERIC: This can be made a generic function in the reference file by passing references for input and output instead of vx_array, and add a third parameter which is the vx_enum for type.